lib/kafo/kafo_configure.rb

```diff
@@ -155,9 +155,44 @@ def initialize(*args)
 
       self.class.hooking.execute(:init)
       set_parameters # here the params gets parsed and we need app config populated
+      migrate_answers_to_user_params_file
       set_options
     end
 
+    def migrate_answers_to_user_params_file
+      return unless config.app[:answer_file]
+      return if File.exist?(config.params_files[ScenarioOption::USER_PARAMS_FILE])
+
+      enabled_modules = config.modules.select(&:enabled?)
+      default_params = Hash[enabled_modules.map { |mod| [mod.identifier, mod.default_params_hash] }]
+      default_params = default_params.reject { |key, value| !value.is_a?(Hash) }
+
+      user_params = Hash[config.modules.map { |mod| [mod.identifier, {}] }]
+
+      config.answers.each do |puppet_class, values|
+        if values.is_a?(Hash)
+          values.each do |param, value|
+            user_params[puppet_class][param] = value unless default_params[puppet_class][param] == value
+          end
+        else
+          user_params[puppet_class] = values unless default_params[puppet_class] == values
+        end
+      end
+
+      config.scenario_params.each do |puppet_class, values|
+        if values.is_a?(Hash)
+          values.each do |param, value|
+            user_params[puppet_class].delete(param) if user_params[puppet_class][param] == value
+          end
+        else
+          user_params.delete(puppet_class) if user_params[puppet_class] == values
+        end
+      end
+
+      config.store(user_params, config.params_files[ScenarioOption::USER_PARAMS_FILE])
+      config.app[:answer_file].delete
```

Comment:
Remove the notion of the answers file from the config but do not delete the answers file. We can leave that on the system as a safety mechanism and user reference.

Reply:
Does that mean that there remains an answers file on disk, which over time gets out of date, as we don't update it anymore? Or do I miss-read the code and it's still getting updated.

I'm mostly concerned that -- while we preach that the file is not to be consumed -- it is used by other tools to look things up, and could lead to unexpected results.

If it's not getting updated anymore, I think we should consider renaming it to something like `asnwers.yml.obsolete` or so, so that users still can manually read it, but tools that expect a certain path, fail explicitly.